### PR TITLE
Chore/ts migrate

### PR DESCRIPTION
**Pull request by kaisermann**
#### What problem is this solving?

Initial migration to TypeScript started by @victorhmp by using https://github.com/airbnb/ts-migrate.

As discussed in our daily at (01/10/2020), initially I was thinking of having a single gigantic PR with all the correct typescript code. However, this will be a lot difficult to make in a single PR, as we have to fix errors in other projects as well. This branch was linked and tested in the workspace mentioned below. After merging, the next steps are smaller prs fixing the `// @ts-expect-error` comments we have throughout the whole codebase.

#### How to test it?

https://tsmigrate--storecomponents.myvtex.com/?

### CODE REVIEW

**Review by guerreirobeatriz — APPROVED** [review]

**Review by iaronaraujo — COMMENTED** [review]
Could this break if the previous proptypes didn't have all the props?
Also, the migration tool left a bunch of comments, it would be nice to check if we can remove them.

File: react/__tests__/components/Container.test.tsx [hunk 1]
```diff
@@ -4,19 +4,21 @@ import { render } from '@vtex/test-tools/react'
 import Container from '../../Container'
 
 describe('<Container /> component', () => {
-  const renderComponent = customProps => {
+  const renderComponent = (customProps: any) => {
```
Comment (klzns) [inline react/__tests__/components/Container.test.tsx:7]:
```suggestion
  const renderComponent = (customProps?: any) => {
```

File: react/__tests__/components/CategoriesHighlights.test.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { render } from '@vtex/test-tools/react'
 import CategoriesHighlights from '../../CategoriesHighlights'
 
 describe('<CategoriesHighlights />', () => {
-  const renderComponent = customProps => {
+  const renderComponent = (customProps: any) => {
```
Comment (klzns) [inline react/__tests__/components/CategoriesHighlights.test.tsx:7]:
```suggestion
  const renderComponent = (customProps?: any) => {
```

File: react/__tests__/components/CategoriesHighlights.test.tsx [hunk 3]
```diff
@@ -15,13 +15,15 @@ describe('<CategoriesHighlights />', () => {
   }
 
   it('should be rendered', () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/CategoriesHighlights.test.tsx:18]: Remove this if suggestion above is accepted
```suggestion
```

File: react/__tests__/components/CategoriesHighlights.test.tsx [hunk 4]
```diff
@@ -15,13 +15,15 @@ describe('<CategoriesHighlights />', () => {
...
     const { asFragment } = renderComponent()
 
     expect(asFragment()).toBeDefined()
     expect(asFragment()).toBeTruthy()
   })
 
   it('should match snapshot Squared', () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/CategoriesHighlights.test.tsx:26]:
```suggestion
```

File: react/__tests__/components/BuyButton.test.tsx [hunk 5]
```diff
@@ -19,7 +20,7 @@ const mocks = [
 ]
 
 describe('<BuyButton />', () => {
-  const renderComponent = (customProps, text = 'Test') => {
+  const renderComponent = (customProps: any, text = 'Test') => {
```
Comment (klzns) [inline react/__tests__/components/BuyButton.test.tsx:23]: Add optional for the second argument too
```suggestion
  const renderComponent = (customProps?: any, text = 'Test') => {
```

File: react/__tests__/components/BuyButton.test.tsx [hunk 6]
```diff
@@ -31,6 +32,7 @@ describe('<BuyButton />', () => {
   }
 
   it('should be rendered', async () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1-2 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/BuyButton.test.tsx:35]:
```suggestion
```

File: react/__tests__/components/Container.test.tsx [hunk 7]
```diff
@@ -4,19 +4,21 @@ import { render } from '@vtex/test-tools/react'
...
 
   it('should be rendered', () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
     const { asFragment } = renderComponent()
 
     expect(asFragment()).toBeTruthy()
   })
 
   it('should match snapshot', () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/Container.test.tsx:21]:
```suggestion
```

File: react/__tests__/components/Container.test.tsx [hunk 8]
```diff
@@ -4,19 +4,21 @@ import { render } from '@vtex/test-tools/react'
...
 describe('<Container /> component', () => {
-  const renderComponent = customProps => {
+  const renderComponent = (customProps: any) => {
     const comp = <Container {...customProps}> Test </Container>
 
     return render(comp)
   }
 
   it('should be rendered', () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/Container.test.tsx:14]:
```suggestion
```

File: react/__tests__/components/Greeting.test.tsx [hunk 9]
```diff
@@ -24,7 +25,7 @@ describe('<Greeting /> component', () => {
     jest.useFakeTimers()
   })
 
-  const renderComponent = customProps => {
+  const renderComponent = (customProps: any) => {
```
Comment (klzns) [inline react/__tests__/components/Greeting.test.tsx:28]:
```suggestion
  const renderComponent = (customProps?: any) => {
```

File: react/__tests__/components/Greeting.test.tsx [hunk 10]
```diff
@@ -33,6 +34,7 @@ describe('<Greeting /> component', () => {
   }
 
   it('should render name in orderForm', async () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/Greeting.test.tsx:37]:
```suggestion
```

File: react/__tests__/components/InfoCard.test.tsx [hunk 11]
```diff
@@ -4,7 +4,7 @@ import { render } from '@vtex/test-tools/react'
 import InfoCard from '../../InfoCard'
 
 describe('<InfoCard />', () => {
-  const renderComponent = props => {
+  const renderComponent = (props: any) => {
```
Comment (klzns) [inline react/__tests__/components/InfoCard.test.tsx:7]:
```suggestion
  const renderComponent = (props?: any) => {
```

File: react/__tests__/components/InfoCard.test.tsx [hunk 12]
```diff
@@ -29,6 +29,7 @@ describe('<InfoCard />', () => {
     expect(asFragment()).toMatchSnapshot()
   })
   it('should render with image and text side by side', () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/InfoCard.test.tsx:32]:
```suggestion
```

File: react/__tests__/components/Logo.test.tsx [hunk 13]
```diff
@@ -4,7 +4,7 @@ import { render } from '@vtex/test-tools/react'
 import Logo from '../../Logo'
 
 describe('<Logo /> component', () => {
-  const renderComponent = customProps => {
+  const renderComponent = (customProps: any) => {
```
Comment (klzns) [inline react/__tests__/components/Logo.test.tsx:7]:
```suggestion
  const renderComponent = (customProps?: any) => {
```

File: react/__tests__/components/Logo.test.tsx [hunk 14]
```diff
@@ -22,6 +22,7 @@ describe('<Logo /> component', () => {
   })
 
   it('should match snapshot without link', () => {
+    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
```
Comment (klzns) [inline react/__tests__/components/Logo.test.tsx:25]:
```suggestion
```

File: react/__tests__/components/ProductDescription.test.tsx [hunk 15]
```diff
@@ -4,7 +4,7 @@ import { render } from '@vtex/test-tools/react'
 import ProductDescription from '../../ProductDescription'
 
 describe('<ProductDescription />', () => {
-  const renderComponent = customProps => {
+  const renderComponent = (customProps: any) => {
```
Comment (klzns) [inline react/__tests__/components/ProductDescription.test.tsx:7]:
```suggestion
  const renderComponent = (customProps?: any) => {
```

File: react/__tests__/components/ProductName.test.tsx [hunk 16]
```diff
@@ -12,13 +12,14 @@ describe('<ProductName />', () => {
 
   const context = { culture: { currency: 'USD' } }
 
-  const renderComponent = customProps => {
+  const renderComponent = (customProps: any) => {
```
Comment (klzns) [inline react/__tests__/components/ProductName.test.tsx:15]: same

File: react/components/ShippingSimulator/components/ShippingTable.tsx [hunk 17]
```diff
@@ -1,24 +1,40 @@
 import React from 'react'
-import PropTypes from 'prop-types'
 import { FormattedMessage } from 'react-intl'
 
 import ShippingTableRow from './ShippingTableRow'
 import styles from '../shippingSimulator.css'
 
-const ShippingTable = ({ shipping }) => {
+type Props = {
```
Comment (klzns) [inline react/components/ShippingSimulator/components/ShippingTable.tsx:7]:
```suggestion
interface Props {
```

**Review by klzns — CHANGES_REQUESTED** [review]

**Comment by kaisermann** [comment]
Will do this one by one

**Pull request closed without merging**